ext/opcache/jit/zend_jit_internal.h

```diff
@@ -717,14 +717,21 @@ static zend_always_inline const zend_op* zend_jit_trace_get_exit_opline(zend_jit
 	return NULL;
 }
 
-static zend_always_inline bool zend_jit_may_be_polymorphic_call(const zend_op *opline)
+static zend_always_inline bool zend_jit_may_be_polymorphic_call(const zend_op *opline, const zend_function *func)
 {
-	if (opline->opcode == ZEND_INIT_FCALL
-	 || opline->opcode == ZEND_INIT_FCALL_BY_NAME
-	 || opline->opcode == ZEND_INIT_NS_FCALL_BY_NAME) {
+	if (opline->opcode == ZEND_INIT_FCALL) {
+		return 0;
+	}
```

Comment:
INIT_FCALL is used when the caller and callee are declared unconditionally in the same file, so if func changes it implies that the caller changed too -> we can assume that func never changes.